lib/terminal-table/table.rb

```diff
@@ -155,6 +155,11 @@ def title=(title)
       recalc_column_widths Row.new(self, [title_cell_options])
     end
 
+    # a table is empty if its rows have no useful content (separators aside)
+    def empty?
+      rows.empty?
```

Comment:
I don't think so. Rows are initialized as an empty array by default:

```
self.rows = options.fetch :rows, []
```

Pretty much the same in the setter.